Test failure with scipy 1.4.0/1.4.1

The latest release of scipy introduced a new `scipy.fft` module, and tests are failing, which is maybe related: https://travis-ci.org/musevlt/origin/jobs/628661877

```>       orig.step05_compute_TGLR(ncpu=1)

tests/test_origin.py:99: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

muse_origin/steps.py:273: in __call__

    self.run(self.orig, *args, **kwargs)

muse_origin/steps.py:777: in run

    pmeansub=pmeansub,

muse_origin/lib_origin.py:68: in timed

    result = f(*args, **kw)

muse_origin/lib_origin.py:1131: in Correlation_GLR_test

    for i in range(Nz)

.tox/py36-cov/lib/python3.6/site-packages/joblib/parallel.py:1004: in __call__

    if self.dispatch_one_batch(iterator):

.tox/py36-cov/lib/python3.6/site-packages/joblib/parallel.py:835: in dispatch_one_batch

    self._dispatch(tasks)

.tox/py36-cov/lib/python3.6/site-packages/joblib/parallel.py:754: in _dispatch

    job = self._backend.apply_async(batch, callback=cb)

.tox/py36-cov/lib/python3.6/site-packages/joblib/_parallel_backends.py:209: in apply_async

    result = ImmediateResult(func)

.tox/py36-cov/lib/python3.6/site-packages/joblib/_parallel_backends.py:590: in __init__

    self.results = batch()

.tox/py36-cov/lib/python3.6/site-packages/joblib/parallel.py:256: in __call__

    for func, args, kwargs in self.items]

.tox/py36-cov/lib/python3.6/site-packages/joblib/parallel.py:256: in <listcomp>

    for func, args, kwargs in self.items]

muse_origin/lib_origin.py:1034: in _convolve_fsf

    cube_fsf = fftconvolve(cube, psf, mode='same')

.tox/py36-cov/lib/python3.6/site-packages/scipy/signal/signaltools.py:542: in fftconvolve

    ret = _freq_domain_conv(in1, in2, axes, shape, calc_fast_len=True)

.tox/py36-cov/lib/python3.6/site-packages/scipy/signal/signaltools.py:382: in _freq_domain_conv

    sp1 = fft(in1, fshape, axes=axes)

.tox/py36-cov/lib/python3.6/site-packages/scipy/fft/_backend.py:23: in __ua_function__

    return fn(*args, **kwargs)

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

forward = True

x = array([[-0.37443433,  1.17728097,  0.17462521, ..., -0.92230921,

         0.15581388, -1.08591093],

       [ 0.1879498...332,  2.01586593],

       [ 0.36455996,  0.70848931, -0.01467468, ...,  0.3368053 ,

         1.820535  ,  0.31100637]])

s = [90, 108], axes = [0, 1], norm = 0, overwrite_x = False, workers = 1

    def r2cn(forward, x, s=None, axes=None, norm=None, overwrite_x=False,

             workers=None):

        """Return multi-dimensional discrete Fourier transform of real input"""

        tmp = _asfarray(x)

    

        if not np.isrealobj(tmp):

            raise TypeError("x must be a real sequence")

    

        shape, axes = _init_nd_shape_and_axes(tmp, s, axes)

        tmp, _ = _fix_shape(tmp, shape, axes)

        norm = _normalization(norm, forward)

        workers = _workers(workers)

    

        if len(axes) == 0:

            raise ValueError("at least 1 axis must be transformed")

    

        # Note: overwrite_x is not utilised

>       return pfft.r2c(tmp, axes, forward, norm, None, workers)

E       RuntimeError: unsupported data type

```

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,7 @@ branches:
 python:
   - "3.6"
   - "3.7"
+  - "3.8"
 
 install: pip install -U tox-travis
 

--- a/muse_origin/lib_origin.py
+++ b/muse_origin/lib_origin.py
@@ -8,7 +8,7 @@ from functools import wraps
 from time import time
 
 #import warnings
-#warnings.filterwarnings("ignore", category=RuntimeWarning) 
+#warnings.filterwarnings("ignore", category=RuntimeWarning)
 
 import matplotlib.pyplot as plt
 import numpy as np
@@ -1120,6 +1120,10 @@ def Correlation_GLR_test(
         # (maybe because of astropy.io.fits doing weird things with the memap?)
         cube = np.array(cube)
 
+    # Make sure that we have a float array in C-order because scipy.fft
+    # (new in v1.4) fails with Fortran ordered arrays.
+    cube = cube.astype(float)
+
     with Parallel(n_jobs=nthreads) as parallel:
         for nf in fields:
             # convolve spatially each spectral channel by the FSF, and do the
@@ -1992,7 +1996,7 @@ def unique_sources(table):
       cube before the PCA.
     - line_merged_flag: boolean flag indicating if any of the lines associated
       to the source was merged with another nearby line.
-    - waves: a list of the first three wavelengths (comma separated), sorted by decreasing flux 
+    - waves: a list of the first three wavelengths (comma separated), sorted by decreasing flux
 
     Note: The n_lines contains the number of unique lines associated to the
     source, but for computing the position of the source, we are using all the
@@ -2032,8 +2036,8 @@ def unique_sources(table):
         seg_label = group['seg_label'][0]
         comp = group['comp'][0]  # FIXME: not necessarily true
         line_merged_flag = np.any(group["line_merged_flag"])
- 
-        ngroup = group[group['merged_in'] == -9999]       
+
+        ngroup = group[group['merged_in'] == -9999]
         ngroup.sort('flux')
         waves = ','.join([str(int(l)) for l in ngroup['lbda'][:-4:-1]])
 

--- a/tox.ini
+++ b/tox.ini
@@ -1,10 +1,11 @@
 [tox]
-envlist = {py36,py37}{,-cov,-mpdafdev},docs
+envlist = py{36,37,38}{,-cov,-mpdafdev},docs
 
 [travis]
 python =
   3.6: py36-cov
   3.7: py37-cov
+  3.8: py38-cov
 
 [testenv]
 whitelist_externals = which

